### PR TITLE
Set display in bgfx platform data

### PR DESCRIPTION
**Pull request by guusw**
Fixes a crash when creating a BGFX window on linux.
It's calling XGetXCBConnection(nullptr) if the display is not set in the platform data

Simplified SDL_GetNativeWindowPtr definition a bit to remove the specific types, since they're not used anyways

### CODE REVIEW

File: src/extra/bgfx.cpp [hunk 1]
```diff
@@ -774,6 +768,7 @@ struct MainWindow : public BaseWindow {
     // set platform data this way.. or we will have issues if we re-init bgfx
     bgfx::PlatformData pdata{};
     pdata.nwh = _sysWnd;
+    // pdata.ndt = SDL_GetNativeDisplayPtr(_window);
```
Comment (sinkingsugar) [inline src/extra/bgfx.cpp:771]: how about this?

**Review by sinkingsugar — APPROVED** [review]
Cool, I think there is just an extra comment to remove.

File: src/extra/bgfx.cpp (hunk 1 above)
Comment (guusw) [reply src/extra/bgfx.cpp:771]: 😅 don't know how I managed to forget the actual fix, thanks

**Comment by sinkingsugar** [comment]
Does this fix sporadic linux CI failure too ? Not sure though , as that is more VM related maybe

**Comment by guusw** [comment]
Don't think so, I think I encountered this because I built linux with the vulkan BGFX backend.

**Pull request merged**